src/metadata-parser.h

```diff
@@ -321,6 +322,52 @@ namespace librealsense
         }
     };
 
+    template<class S, class Attribute, typename Flag>
+    class md_attribute_parser_with_crc : public md_attribute_parser<S, Attribute, Flag>
+    {
+    public: 
+        md_attribute_parser_with_crc(Attribute S::* attribute_name, Flag flag, unsigned long long offset, attrib_modifyer mod, unsigned long long crc_offset)
+            : md_attribute_parser<S, Attribute, Flag>(attribute_name, flag, offset, mod), _crc_offset(crc_offset) {}
+
+    protected:
+        unsigned long long _crc_offset; // offset of the crc inside the S class
+        bool is_attribute_valid(const S* s) const override
+        {
+            if (!md_attribute_parser<S, Attribute, Flag>::is_attribute_valid(s))
+                return false;
+
+            if (!is_crc_valid(s))
+            {
+                LOG_DEBUG("Metadata CRC mismatch");
```

Comment:
'LOG_WARNING?'

Reply:
This is the way it was originally, and it is also LOG_DEBUG in the parent class in is_attribute_valid function, when we find "metadata mismatch"